src/Neo/SmartContract/NotifyEventArgs.cs

```diff
@@ -69,7 +69,7 @@ public StackItem ToStackItem(ReferenceCounter referenceCounter)
             {
                 ScriptHash.ToArray(),
                 EventName,
-                State
+                State.OnStack ? State : State.DeepCopy(true)
```

Comment:
It seems to me that this change is the one that actually fixes the problem. But the problem is that it requires one more hardfork if we care about safety since it changes the behaviour.

Reply:
just apply to the hardfork you created.  will update.

Reply:
Hardfork has being applied.